Src/Services/ActionService.lua

```diff
@@ -38,6 +42,29 @@ function ActionService.OnStart(self: ActionService)
 	self.InternalActionsLoaded:Set(true)
 end
 
+function ActionService.OnInit(self: ActionService)
+	Action.ActionAdded:Connect(function(actionObject: Action.Action)
+		local camelCaseActionArguments = { }
+
+		if actionObject.Arguments then
+			for index, actionMetadata in next, actionObject.Arguments do
+				camelCaseActionArguments[index] = {
+					argumentDefault = actionMetadata.ArgumentDefault,
+					argumentIsOptional = actionMetadata.ArgumentIsOptional,
+					argumentName = actionMetadata.ArgumentName,
+					argumentType = actionMetadata.ArgumentType
+				}
+			end
+		end
+
+		ApiService:PostAsync(Api.RegisterAction, {
```

Comment:
maybe we could manually not upload it here if it starts with `internal:`

Reply:
sure, will there be an API dedicated for internal actions? 😄

Reply:
Yes I was planning on calling those directly with MessagingService, and the results can be uploaded to api.com/(moderation, broadcasts, etc)/result